### PR TITLE
Adds two new artifact effects: Plant killer and Plant helper

### PR DESCRIPTION
**Pull request by jknpj**
Adds plant helper and plant killer as possible artifact effects. 

Plant Helper aura effect will refill your plants water and nutriment really slowly plus deal with weeds and pests.
Plant Helper pulse effect will refill water, nutriments, remove weeds, pests and toxins and even rejuvenate your crops.

Plant Killer aura effect will dry your crops of both water and nutriment, make them age faster and make weeds appear more often.
Plant Killer pulse effect will quickly remove water and nutriment, fill their trays with toxin and make pests and weeds appear easily. It will also have 5% of instantly killing a target when adding toxins to it.

:cl:
 * rscadd: Adds two new artifact effects: plant helper and plant killer.

### CODE REVIEW

File: code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	..()
+	effect = pick(EFFECT_AURA, EFFECT_PULSE)
+
+/datum/artifact_effect/planthelper/DoEffectAura()
+	if(holder)
+		for (var/obj/machinery/portable_atmospherics/hydroponics/H in range(src.effectrange,holder))
+			if(H.seed && !H.dead && prob(rand(6,12)))
+				switch(rand(1,4))
+					if(1)
+						H.waterlevel += rand(2,8)
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm]: These should probably be capped unless hydroponics code can handle it.

File: code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+/datum/artifact_effect/planthelper/DoEffectPulse()
+	if(holder)
+		for(var/obj/machinery/portable_atmospherics/hydroponics/H in range(src.effectrange,holder))
+			if(H.seed && !H.dead && prob(rand(8,36)))
+				switch(rand(1,4))
+					if(1)
+						H.waterlevel += rand(5,15)
+						H.nutrilevel += rand(1,10)
+					if(2) // This will totally end fine.
+						H.mutation_level++
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm]: Does this even work? Sounds like you'd need to call procs, no?

File: code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+					if(3)
+						H.weedlevel--
+					if(4)
+						H.pestlevel--
+		return 1
+
+/datum/artifact_effect/planthelper/DoEffectPulse()
+	if(holder)
+		for(var/obj/machinery/portable_atmospherics/hydroponics/H in range(src.effectrange,holder))
+			if(H.seed && !H.dead && prob(rand(8,36)))
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm]: > `prob(rand())`

This guy understands how probabilities work.

File: code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+				switch(rand(1,4))
+					if(1)
+						H.waterlevel += rand(5,15)
+						H.nutrilevel += rand(1,10)
+					if(2) // This will totally end fine.
+						H.mutation_level++
+						H.mutation_mod++
+						H.yield_mod++
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm]: This seems extreme.

Just leave this sitting at your plant for a few minutes and the yield will become insane.

File: code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_plantkiller.dm [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	..()
+	effect = pick(EFFECT_AURA, EFFECT_PULSE)
+
+/datum/artifact_effect/plantkiller/DoEffectAura()
+	if(holder)
+		for (var/obj/machinery/portable_atmospherics/hydroponics/H in range(src.effectrange,holder))
+			if(H.seed && !H.dead && prob(rand(5,25)))
+				switch(rand(1,3))
+					if(1)
+						H.waterlevel -= rand(1,5)
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_plantkiller.dm]: Caps.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
The bad mutation possibility for the helper is probably not something that should be included.

File: code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm (hunk 1 above)
Comment (jknpj) [reply code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm]: max water is capped at 100

File: code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm (hunk 3 above)
Comment (Aurx) [reply code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_planthelper.dm]: It's extra random!

**Comment by AI-Is-Rogue** [comment]
Is there any reason someone would want the plant killer effect?

**Comment by Shadowmech88** [comment]
Not all artifact effects need to be desirable.

**Comment by AI-Is-Rogue** [comment]
I know, but I am still curious.

**Comment by jknpj** [comment]
@AI-is-rogue https://www.youtube.com/watch?v=ugJfjmxOR2I

**Comment by N3X15** [comment]
@PJB3005 You done with this?

**Review by PJB3005 — APPROVED** [review]

**Comment by FudgePucker** [comment]
I just came up with an idea for the plant killer effect, make it work on kudzu.

**Pull request merged**